ml_genn/ml_genn/compilers/event_prop_compiler.py

```diff
@@ -970,50 +972,86 @@ def create_compiled_network(self, genn_model, neuron_populations: dict,
+                            o, genn_model,
+                            (0.0, c.max_delay_steps))
+
+                        # Add custom update to list of optimisers
+                        optimisers.append((deepcopy(o), cu_delay))
                 
-                # Add gradient to list of gradient vars to zero
-                gradient_vars.append(("delayGradient", "scalar", 0.0))
-
-            # Create reset model for gradient variables
-            assert len(gradient_vars) > 0
-            zero_grad_model = create_reset_custom_update(
-                gradient_vars,
-                lambda name: create_wu_var_ref(genn_pop, name))
-
-            # Add custom update
-            self.add_custom_update(genn_model, zero_grad_model, 
-                                   "ZeroGradient", f"CUZeroConnGradient{i}")
+                        # Add gradient to list of gradient vars to zero
+                        gradient_vars.append(("delayGradient", "scalar", 0.0))
+
+                # Create reset model for gradient variables
+                assert len(gradient_vars) > 0
+                zero_grad_model = create_reset_custom_update(
+                    gradient_vars,
+                    lambda name: create_wu_var_ref(genn_pop, name))
+
+                # Add custom update
+                self.add_custom_update(genn_model, zero_grad_model,
+                                       "ZeroGradient", f"CUZeroConnGradient{i}")
+                need_zero_gradient_update_group = True
+                i = i+1
+
+        # Loop through populations that require optimisers
+        i = 0
+        for c, optim in compile_state.optimisers.items():
+            if c in neuron_populations:
+                genn_pop = neuron_populations[c]
+                #gradient_vars = []
+                for p, o in optim.items():
+                    # Create parameter optimiser custom update
+                    cu_param = self._create_optimiser_custom_update(
+                        f"{p}{i}", create_var_ref(genn_pop, p),
+                            create_var_ref(genn_pop, f"{p}Gradient"),
+                            o, genn_model)
+
+                    # Add custom update to list of optimisers
+                    optimisers.append((deepcopy(o), cu_param))
+
+                    # Add gradient to list of gradient vars to zero
+                    # JAMIE: is this necessary or am I duplicating reset_vars
+                    #gradient_vars.append((f"{p}Gradient", "scalar", 0.0))
+                # Create reset model for gradient variables
+                #assert len(gradient_vars) > 0
+                #zero_grad_model = create_reset_custom_update(
+                #    gradient_vars,
+                #    lambda name: create_var_ref(genn_pop, name))
+
+                # Add custom update
+                #self.add_custom_update(genn_model, zero_grad_model,
+                #                       "ZeroGradient", f"CUZeroPopGradient{i}")
```

Comment:
Yes, this is correct - the normal resetting mechanism should handle this correctly for population variables

```suggestion
```